docs(dotnet): add Data Collected pages (#13529)
getsentry/sentry-docs: docs/platforms/unity/data-management/data-collected.mdx

## HTTP Headers

By default, the Sentry SDK sends HTTP headers to Sentry but redacts any headers that may contain sensitive data. (See the [list of headers](https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/Internal/RedactedHeaders.cs#L7) that are redacted).

## Cookies

By default, the Sentry SDK doesn't send cookies.

If you want to send cookies, set the option `SendDefaultPii = true` when initializing the Sentry SDK.

A [data scrubber](/platforms/dotnet/data-management/sensitive-data/) can be used to filter information before sending data to Sentry.

## Information About Logged-in User

By default, the Sentry SDK doesn't send any information about the logged-in user, such as email address, user ID, or username. Even if enabled, the type of logged-in user information you'll be able to send depends on the integrations you enable in Sentry's SDK. Most integrations won't send any user information. Some will only set the user ID, but there are a few that will set the user ID, username, and email address.

> To start sending logged-in user information, set the option `SendDefaultPii = true` when initializing the Sentry SDK.

Reviewer: Not sure Unity adds User if we flip this flag. @bitsandfoxes ?

Reviewer 2: The Unity SDK guards and sets the userID [here](https://github.com/getsentry/sentry-unity/blob/21c9d563f2d75e7cbe64061321d0fa65442b30d3/src/Sentry.Unity/Integrations/UnityScopeIntegration.cs#L172-L181).
On Desktop and WebGL, the `options.defaultUserId` resolves to Unity's [AnalyticsSessionInfo.userId](https://docs.unity3d.com/2018.3/Documentation/ScriptReference/Analytics.AnalyticsSessionInfo-userId.html). 

On mobile the SDK tries to fetch the `Installation ID` first and if it fails, it falls back to AnalyticsSessionInfo.userId as well.

Reviewer 2: It does not. The one place the Unity SDK uses `SendDefaultPii` for guarding and setting the `DeviceUniqueIdentifier` [here](https://github.com/getsentry/sentry-unity/blob/21c9d563f2d75e7cbe64061321d0fa65442b30d3/src/Sentry.Unity/Integrations/UnityScopeIntegration.cs#L92-L94). The rest is in control of the .NET SDK.

## Local Variables In Stack Trace

When unhandled errors and exceptions are captured, a stack trace including the names and values of local variables at the time of the error will be sent to Sentry.

## SQL Queries

While SQL queries are sent to Sentry, neither the full SQL query (`SELECT * FROM users WHERE email = 'joe@shmoe.com'`), nor the values of its parameters will ever be sent. A parameterized version of the query (`SELECT * FROM users WHERE email = @email`) is sent instead.

## Machine Name

The Sentry SDK does not send the machine name by default.

If you assign a value to the `ServerName` in the Sentry options then Sentry SDK will send this with any events.

Otherwise, if `SendDefaultPii` is set to true then the Sentry SDK will capture the `Environment.MachineName` automatically and send this with any events.

## General Device Information

The Sentry SDK collects information about the device, such as the version and build of your operating system, CPU and GPU details etc. This information is sent to Sentry by default.

> ## Data captured by Mobile SDKs
> 
> On mobile platforms, the Sentry SDK for .NET can also capture data indirectly via the Android and iOS SDKs. You should refer to the documentation for those SDKs for more information on what data is captured.
> 
> - [Android](/platforms/android/data-management/data-collected/)
> - [iOS](/platforms/apple/data-management/data-collected/)
> 
> ## Native Crash Reporting
> 
> When compiling .NET applications Ahead-Of-Time (AOT), native crashes may also be captured. You should refer to [data captured by the Native SDK](/platforms/native/data-management/data-collected/) to understand what data is sent with those events.

Reviewer: Sentry Native here is not related to Native AOT (they use IL2CPP and use it on all platforms potentially, including where we don't have Sentry Native but Android/Cocoa instead.

```suggestion
## Data captured by Mobile SDKs

On mobile platforms, the Sentry SDK for .NET can also capture data indirectly via the Android and iOS SDKs. You should refer to the documentation for those SDKs for more information on what data is captured.

- [Android](/platforms/android/data-management/data-collected/)
- [iOS](/platforms/apple/data-management/data-collected/)

## Data captured by the Native SDKL

When compiling for Windows, Linux or other platforms that use Sentry Native, you should refer to [data captured by the Native SDK](/platforms/native/data-management/data-collected/) to understand what data is sent with those events.
```

## Cookies

By default, the Sentry SDK doesn't send cookies.

If you want to send cookies, set the option `SendDefaultPii = true` when initializing the Sentry SDK.

A [data scrubber](/platforms/dotnet/data-management/sensitive-data/) can be used to filter information before sending data to Sentry.

Any [sensitive cookies](https://github.com/getsentry/relay/blob/249971e1702991c64cf0f7f89773fb019ffb4a95/relay-pii/src/convert.rs#L235-L263) that the SDK does send will be stripped by Relay before Sentry stores these.

## Information About Logged-in User

By default, the Sentry SDK doesn't send any information about the logged-in user, such as email address, user ID, or username. Even if enabled, the type of logged-in user information you'll be able to send depends on the integrations you enable in Sentry's SDK. Most integrations won't send any user information. Some will only set the user ID, but there are a few that will set the user ID, username, and email address.

> To start sending logged-in user information, set the option `SendDefaultPii = true` when initializing the Sentry SDK.

Reviewer: ```suggestion
To start sending logged-in user information, set the option `SendDefaultPii = true` when initializing the Sentry SDK. This will leverage Unity's [DeviceUniqueIdentifier](https://docs.unity3d.com/Documentation/ScriptReference/SystemInfo-deviceUniqueIdentifier.html)
```

## Request URL

By default, the [HttpRequestUrl](https://learn.microsoft.com/en-us/dotnet/api/System.UriComponents) of outgoing and incoming HTTP requests is **always sent to Sentry**. This includes the Scheme, Host, Port, LocalPath, and Query data.

If `SendDefaultPii` is set to `true` then the [AbsoluteUri](https://learn.microsoft.com/en-us/dotnet/api/System.UriComponents) of outgoing and incoming HTTP requests will be sent to Sentry (including the Scheme, UserInfo, Host, Port, LocalPath, Query, and Fragment data).

Depending on your application, this could contain PII data.

## Request Query String

The full request query string of outgoing and incoming HTTP requests is **always sent to Sentry**. Depending on your application, this could contain PII data.

## SQL Queries

While SQL queries are sent to Sentry, neither the full SQL query (`SELECT * FROM users WHERE email = 'joe@shmoe.com'`), nor the values of its parameters will ever be sent. A parameterized version of the query (`SELECT * FROM users WHERE email = @email`) is sent instead.

> 

Reviewer 2: ```suggestion

## Device Unique Identifier

The Sentry SDK does not set the `Device Unique Identifier`  by default.

If `SendDefaultPii` is set to `true` then the SDK will capture `SystemInfo.deviceUniqueIdentifier` automatically and send this with any event.
```

## Request URL

By default, the [HttpRequestUrl](https://learn.microsoft.com/en-us/dotnet/api/System.UriComponents) of outgoing and incoming HTTP requests is **always sent to Sentry**. This includes the Scheme, Host, Port, LocalPath, and Query data.

If `SendDefaultPii` is set to `true` then the [AbsoluteUri](https://learn.microsoft.com/en-us/dotnet/api/System.UriComponents) of outgoing and incoming HTTP requests will be sent to Sentry (including the Scheme, UserInfo, Host, Port, LocalPath, Query, and Fragment data).

Depending on your application, this could contain PII data.

## Request Query String

The full request query string of outgoing and incoming HTTP requests is **always sent to Sentry**. Depending on your application, this could contain PII data.

## SQL Queries

While SQL queries are sent to Sentry, neither the full SQL query (`SELECT * FROM users WHERE email = 'joe@shmoe.com'`), nor the values of its parameters will ever be sent. A parameterized version of the query (`SELECT * FROM users WHERE email = @email`) is sent instead.

## Device Unique Identifier

> The Sentry SDK does not set the `Device Unique Identifier`  by default.

Reviewer 2: ```suggestion
The Sentry SDK does not set the `Device Unique Identifier` by default.
```
